Artifacts/windows-vsts-download-and-run-script/DownloadVstsDropAndExecuteScript.ps1

```diff
@@ -1,139 +1,201 @@
+    $artifacts = Invoke-RestMethod -Uri $ArtifactsUri -Headers $Headers -Method Get | ConvertTo-Json -Depth 3 | ConvertFrom-Json
+    $downloadUrl = $artifacts.value.resource.downloadUrl
+    
+    Write-Host "Downloading build artifacts package from $downloadUrl"
+    Invoke-RestMethod -Uri $downloadUrl -Headers $Headers -Method Get -Outfile $Outfile | Out-Null
 
-        [System.Reflection.Assembly]::LoadWithPartialName("System.IO.Compression.FileSystem") | Out-Null 
-        [System.IO.Compression.ZipFile]::ExtractToDirectory($outfile, $destination)
-    }
-    catch
+    if (Test-Path $Destination -PathType Container)
     {
-        if (($null -ne $Error[0]) -and ($null -ne $Error[0].Exception) -and ($null -ne $Error[0].Exception.Message))
-        {
-            $errMsg = $Error[0].Exception.Message
-            Write-Host $errMsg
-        }
-        exit -1
+        Write-Host "Cleaning up destination $Destination"
+        Remove-Item -Path $Destination -Force -Recurse | Out-Null
     }
+
+    Write-Host "Extracting build artifacts package content to $Destination"
+    [System.Reflection.Assembly]::LoadWithPartialName("System.IO.Compression.FileSystem") | Out-Null 
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($Outfile, $Destination) | Out-Null
 }
  
-function RunScript
+function Run-Script
 {
-    $scriptPath = Join-Path -Path $destination -ChildPath $pathToScript 
+    [CmdletBinding()]
+    param (
+        [string] $Path,
+        [string] $Script,
+        [string] $Arguments
+    )
+
+    $scriptPath = Join-Path -Path $Path -ChildPath $Script
 
-    Write-Output $scriptPath
+    Write-Host "Running $scriptPath"
 
     if (Test-Path $scriptPath -PathType Leaf)
     {
-        Invoke-Expression  "& `"$scriptPath`" $scriptArguments"
+        Invoke-Expression "& `"$scriptPath`" $scriptArguments"
     }
+    else
+    {
+        Write-Error "Unable to locate $scriptPath"
+    }
+}
+
+###################################################################################################
+
+#
+# Handle all errors in this script.
+#
+
+trap
+{
+    # NOTE: This trap will handle all errors. There should be no need to use a catch below in this
+    #       script, unless you want to ignore a specific error.
+    Handle-LastError
 }
 
-DownloadBuildArtifacts
-RunScript
+###################################################################################################
+
+#
+# Main execution block.
+#
 
+try
+{
+    # Prepare values used throughout.
+    $vstsApiVersion = "2.0"
+    $outfile = "$PSScriptRoot\$buildDefinitionName.zip"
+    $destination = "$($env:HOMEDRIVE)\$buildDefinitionName"
+    $vstsProjectUri = $vstsProjectUri.TrimEnd("/")
+    $headers = Set-AuthHeaders -UserName $username -AccessToken $accessToken
+
+    # Output provided parameters.
+    Write-Host 'Provided parameters used in this script:'
+    Write-Host "  `$username = $username"
+    Write-Host "  `$accessToken = ************"

```

Comment:
Just following same pattern build uses. I could just remove it to avoid confusion.